File: LAB 4/Scan-Converting_a_Character-Bitmap_Font.py
import matplotlib.pyplot as plt
import numpy as np

plt.figure()
plt.grid(True)
plt.title('Bitmap Font - Character A')
plt.xlabel('X-axis')
plt.ylabel('Y-axis')

CHAR_A = [
    [0, 0, 1, 1, 1, 0, 0],
    [0, 1, 0, 0, 0, 1, 0],
    [1, 0, 0, 0, 0, 0, 1],
    [1, 0, 0, 0, 0, 0, 1],
    [1, 1, 1, 1, 1, 1, 1],
    [1, 0, 0, 0, 0, 0, 1],
    [1, 0, 0, 0, 0, 0, 1]
]

X_VALS = []
Y_VALS = []

for Y in range(len(CHAR_A)):
    for X in range(len(CHAR_A[0])):
        if CHAR_A[Y][X] == 1:
            X_VALS.append(X)
            Y_VALS.append(len(CHAR_A) - Y - 1)

plt.plot(X_VALS, Y_VALS, 'ks', markersize=20, markerfacecolor='k')
plt.xlim(-1, 8)
plt.ylim(-1, 8)
plt.show()

File: LAB 4/MATLAB/Scan-Converting_a_Character-Bitmap_Font.m
clc; clear; close all;

figure;
hold on;
grid on;
axis equal;
title('Bitmap Font - Character A');
xlabel('X-axis');
ylabel('Y-axis');
xlim([-1, 8]);
ylim([-1, 8]);

CHAR_A = [
    0, 0, 1, 1, 1, 0, 0;
    0, 1, 0, 0, 0, 1, 0;
    1, 0, 0, 0, 0, 0, 1;
    1, 0, 0, 0, 0, 0, 1;
    1, 1, 1, 1, 1, 1, 1;
    1, 0, 0, 0, 0, 0, 1;
    1, 0, 0, 0, 0, 0, 1
];

X_VALS = [];
Y_VALS = [];

[ROWS, COLS] = size(CHAR_A);
for Y = 1:ROWS
    for X = 1:COLS
        if CHAR_A(Y, X) == 1
            X_VALS = [X_VALS X-1];
            Y_VALS = [Y_VALS ROWS-Y];
        end
    end
end

plot(X_VALS, Y_VALS, 'ks', 'MarkerSize', 20, 'MarkerFaceColor', 'k');
